lib/puppet/resource/type.rb

```diff
@@ -413,6 +413,9 @@ def assign_defaults(resource, scope, resource_hash)
 
   def validate_resource_hash(resource, resource_hash)
     Puppet::Pops::Types::TypeMismatchDescriber.validate_parameters(resource.to_s, parameter_struct, resource_hash)
+  rescue Puppet::ParseError => e
```

Comment:
It doesn't catch "any problem". It will just catch the case when a parameter expects a value, the environment compiler is on the stack, and the 'consume' metaparameter is present. It will re-raise the error for all other cases.

But this IS indeed a dirty fix. The real fix IMO would be to do what you suggest in the JIRA and I've suggested earlier, to use real types in the type system to describe the capabilities. Not sure I see the point in elaborating for less dirtyness.